### PR TITLE
Centralize definition of brand identifiers

### PR DESCRIPTION
**Pull request by jugglinmike**
Prior to this commit, the name of the Automation Voice and its "vendor"
were defined on an ad-hoc basis. This approach made it difficult to
understand how the brand information was used across the Visual Studio
projects, and this in turn made it difficult to infer and maintain
relationships between the values.

Introduce a Visual Studio "Shared Project" to define all brand
identifiers and reference those definitions consistently.

### CODE REVIEW

**Comment by jugglinmike** [comment]
@mzgoddard I think this only nominally uses the "Shared Project" Visual Studio feature. The intended use case seems to involve adding "References" to the Projects which rely on the shared code, but I didn't add any such References to the existing projects. Instead, the "include" directives use relative paths, and I believe this would work for any arbitrary Visual Studio project. Have you used this feature before? Do you see any reason for us to use it more idiomatically?

File: src/Shared/branding.h [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#pragma once
+#define AUTOMATION_VOICE_ID "BocoupAutomationVoice"
+#define AUTOMATION_VOICE_NAME "Bocoup Automation Voice"
+#define AUTOMATION_VOICE_VENDOR "Bocoup"
+#define AUTOMATION_VOICE_HOME "C:\\Program Files\\Bocoup Automation Voice"
```
Comment (mzgoddard) [inline src/Shared/branding.h:5]: Should there be a newline at the end of this file?

Comment (mzgoddard) [reply src/Shared/branding.h:5]: This could be `#define AUTOMATION_VOICE_HOME "C:\\Program Files\\" AUTOMATION_VOICE_NAME`. What do you think?

**Review by mzgoddard — APPROVED** [review]
Nice

**Comment by mzgoddard** [comment]
> @mzgoddard I think this only nominally uses the "Shared Project" Visual Studio feature. The intended use case seems to involve adding "References" to the Projects which rely on the shared code, but I didn't add any such References to the existing projects. Instead, the "include" directives use relative paths, and I believe this would work for any arbitrary Visual Studio project. Have you used this feature before? Do you see any reason for us to use it more idiomatically?

I haven't used this feature previously. So I don't think I have opinions on that at the moment.

File: src/Shared/branding.h (hunk 1 above)
Comment (jugglinmike) [reply src/Shared/branding.h:5]: I think it's more important to have the full path in the source code to facilitate text search by future contributors. Being DRY here means folks won't get an answer to the question, "Where is `Program Files\\Bocoup Automation Voice` defined?"

**Comment by jugglinmike** [comment]
Thanks for the review, @mzgoddard. Since gh-3 has been merged, I've updated this patch to apply the new centralized values to the code introduced there. Please use GitHub's "squash and merge" feature to land this.

**Pull request merged**